src/client/client.go

```diff
@@ -296,59 +297,62 @@ func getCertOptionsFromEnv() ([]Option, error) {
 // getUserMachineAddrAndOpts is a helper for NewOnUserMachine that uses
 // environment variables, config files, etc to figure out which address a user
 // running a command should connect to.
-func getUserMachineAddrAndOpts(context *config.Context) (string, []Option, error) {
+func getUserMachineAddrAndOpts(context *config.Context) (*grpcutil.PachdAddress, []Option, error) {
```

Comment:
Should this return a literal `PachdAddress`? (or should `ParsePachdAddress` return a pointer? See below)

Reply:
Given it's just one value, I don't have an opinion either way, and just changed it to use pointers everywhere. I'm not sure if there's some sort of official go guidelines on when to use values vs pointers (if there are please do let me know), but I tend to lean on values when I can because:

1. It reduces the chances of panics from null pointer dereferencing.
2. It reduces the amount of pointer chasing and GC pressure.

Reply:
That makes sense. I found one doc in the golang repo on receiver type https://github.com/golang/go/wiki/CodeReviewComments#receiver-type but that involves slightly different considerations. Either way I think it's not a big deal.

Reply:
Oh man, those heuristics are really helpful. Thanks!